classes/app/ai/history/archive.php

```diff
@@ -17,109 +13,163 @@
 class archive implements interfaces\archive
 {
     private base_factory $factory;
+    private int $userid;
     private int $assignment_id;
-    private int $step;
-    private int $user_id;
-    private int $current_time;
-    private moodle_transaction $transaction;
-    private ?attempt_entity $attempt = null;
+    private string $type;
+    private ?int $submission;
```

Comment:
More specific naming

```suggestion
    private ?int $submission_id;
```